lib/core/mp_default_floor_selector.dart

```diff
@@ -1,59 +1,73 @@
 part of '../mapsindoors.dart';
 
 class MPDefaultFloorSelector extends StatefulWidget with MPFloorSelector {
-  MPDefaultFloorSelector({super.key});
-  final _Forwarder _callForwarder = _Forwarder();
+  const MPDefaultFloorSelector({super.key});
 
   @override
   Widget? getWidget() => this;
+
   @override
   bool get isAutoFloorChangeEnabled => true;
+
   @override
-  set floors(List<MPFloor>? floors) => _callForwarder.setFloors?.call(floors);
-  @override
-  set onFloorSelectionChangedListener(
-      OnFloorSelectionChangedListener listener) {
-    if (_callForwarder.setOnFloorSelectionChangedListener != null) {
-      _callForwarder.setOnFloorSelectionChangedListener?.call(listener);
-    } else {
-      _callForwarder.listener = listener;
-    }
+  set floors(List<MPFloor>? value) {
+    // Find the current state using the widget's key
+    final state = _findCurrentState();
+    state?.floors = value;
   }
 
   @override
-  void setSelectedFloor(MPFloor floor) =>
-      _callForwarder.setSelectedFloor?.call(floor);
+  set onFloorSelectionChangedListener(OnFloorSelectionChangedListener value) {
+    final state = _findCurrentState();
+    state?.onFloorSelectionChangedListener = value;
+  }
+
   @override
-  void setSelectedFloorByFloorIndex(int floorIndex) =>
-      _callForwarder.setSelectedFloorByFloorIndex?.call(floorIndex);
+  void setSelectedFloor(MPFloor floor) {
+    final state = _findCurrentState();
+    state?.setSelectedFloor(floor);
+  }
+
   @override
-  set userPositionFloor(int floorIndex) =>
-      _callForwarder.setUserPositionFloor?.call(floorIndex);
+  void setSelectedFloorByFloorIndex(int floorIndex) {
+    final state = _findCurrentState();
+    state?.setSelectedFloorByFloorIndex(floorIndex);
+  }
+
   @override
-  void show(bool show) => _callForwarder.show?.call(show);
+  set userPositionFloor(int value) {
+    final state = _findCurrentState();
+    state?.userPositionFloor = value;
+  }
+
   @override
-  void zoomLevelChanged(num newZoomLevel) =>
-      _callForwarder.zoomLevelChanged?.call(newZoomLevel);
+  void show(bool show) {
+    final state = _findCurrentState();
+    state?.show(show);
+  }
 
   @override
-  FloorSelectorState createState() => FloorSelectorState();
-}
+  void zoomLevelChanged(num newZoomLevel) {
+    final state = _findCurrentState();
+    state?.zoomLevelChanged(newZoomLevel);
+  }
+
+  /// Safely finds the current state instance via the widget's [GlobalKey].
+  _MPDefaultFloorSelectorState? _findCurrentState() {
```

Comment:
This is needed because the widget itself is extending `MPFloorSelector`. I think it would be more idiomatic for the `_MPDefaultFloorSelectorState` state class to handle all of the data manipulation, but I felt that refactor was a little outside the scope of this PR.